test/e2e/framework/pod/utils.go

```diff
@@ -115,10 +120,13 @@ func GetLinuxLabel() *v1.SELinuxOptions {
 		Level: "s0:c0,c1"}
 }
 
+const nonRootUser = 1000
+
 // GetRestrictedPodSecurityContext returns a minimal restricted pod security context.
 func GetRestrictedPodSecurityContext() *v1.PodSecurityContext {
 	return &v1.PodSecurityContext{
 		RunAsNonRoot:   pointer.BoolPtr(true),
+		RunAsUser:      pointer.Int64(nonRootUser),
```

Comment:
I'm still unsure about this... this isn't really a minimal restricted pod security context, right? shouldn't `runAsNonRoot: true` be sufficient?

Reply:
Part of my goal with these methods was that most e2e test authors can just apply `MixinRestrictedPodSecurity` and not think about the restricted profile at all. If we don't set `RunAsUser` here (and in the mixin) then for all cases other than the pause pod the test author will need to manually set a RunAsUser. They'll probably pick something random that gets cargo-culted around.

What's your argument against this?

Reply:
> If we don't set RunAsUser here (and in the mixin) then for all cases other than the pause pod the test author will need to manually set a RunAsUser

Would they *have* to set it, or do you mean because the container image would have to be built to run with a non-root uid, which is uncommon?

> What's your argument against this?

It would break images that expect to run with a particular non-zero uid at runtime, right?

Reply:
I searched through our test image manifests, and the only ones that set a user are:
- pause: user doesn't matter
- nonroot: all the tests using this override the user anyway
- windows containers: don't support Restricted for now (added TODO)

I'd prefer that the tests which require a specific user are explicit about it, rather than having all the tests that don't have an opinion set something arbitrary.

Reply:
> rather than having all the tests that don't have an opinion set something arbitrary

but… isn't this setting something arbitrary (1000) on all tests that don't have an opinion?

Reply:
The difference is that defaulting it in a helper function makes it more obvious that it's just part of the boilerplate to run the test, not an essential part of what's being tested. This is a general complaint that I have about our e2e tests - they often set a bunch of random fields on the resources, and it's not clear which are core to the test.

Reply:
Is runAsUser generally needed as boilerplate, though? I thought runAsNonRoot was supposed to be sufficient.

If an e2e wanted to exercise `runAsNonRoot` functionality with an image that did not run as root by default and didn't want to set a runAsUser, that would mean they couldn't use GetRestrictedPodSecurityContext or MustMixinRestrictedPodSecurity to do so, because those wouldn't *actually* set the minimal securityContext that is compatible with the restricted PSS.

Reply:
1. IMO, if a test is explicitly testing the runAsNonRoot functionality (with or without a runAsUser set), then the test should be explicitly writing & managing the SecurityContext, and not deferring that to a helper.
2. There should probably be 1-2 E2E tests explicitly checking runAsNonRoot behavior. I'd prefer to optimize these convenience methods for the common path, and force those couple of tests to be write the pod manually.

Reply:
Ok, fair enough. Make sure the docs don't claim to produce a minimal valid restricted securityContext and doc the uid chosen

Reply:
This reasoning makes sense to me 👍  - A comment summarizing this would go a long way for future readers of the framework code here tho.

Reply:
Updated comments.